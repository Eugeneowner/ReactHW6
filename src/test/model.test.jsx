/*
describe("Modal component", () =>{
    it("Check render component", () =>{
        expect(...).toBe(value)
    })
})

*/
import Modal from "../componets/modal/modal";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
describe("Modal component", () => {
  it("Check render component without props", () => {
    render(<Modal />);
    expect(screen.getByText("Confirm")).toBeInTheDocument();
    expect(screen.getByText("Cancel")).toBeInTheDocument();
  });
  it("Check snapshot without props", () => {
    const modal = render(<Modal />);
    expect(modal).toMatchSnapshot();
  })
  it("Check render component with props", () => {
    const modal = {
      title: "title",
      text: "text",
    };
    render(<Modal title={modal.title} text={modal.text} />);
    expect(screen.getByText(modal.title)).toBeInTheDocument();
    expect(screen.getByText(modal.text)).toBeInTheDocument();
  });
  it("Check snapshot with props", () => {
    const modal = render(<Modal />)
    expect(modal).toMatchSnapshot();
  })
  it ("Check render component close modal", () => {
    render(<Modal closeButton={true} />)
    const closeBtn = screen.getByTestId("CloseBtn");
    expect(closeBtn).toBeInTheDocument(); 
  });
  it("Check snapshot component without close", () => {
    const modal = render(<Modal closeButton={false} />)
    expect(modal).toMatchSnapshot();
  })
  it("Check render component without close btn upper", () => {
    render(<Modal closeButton={false} />)
    expect(screen.queryByTestId("CloseBtn")).toBeNull()
  });
  it("Check snapshot component without btn upper", () => {
    const modal = render(<Modal closeButton={false} />)
    expect(modal).toMatchSnapshot();
  })
  it("Check click on buttons confirm click", () => {
    const onClick = jest.fn();
    render(<Modal handleConfirm={onClick} />);
    const button = screen.getByText("Confirm");
    userEvent.click(button);
    expect(onClick).toHaveBeenCalledTimes(1);
  });
  it("Check click on buttons cancel click", () => {
    const onClick = jest.fn();
    render(<Modal handleClickClose={onClick} />);
    const button = screen.getByText("Cancel");
    userEvent.click(button);
    expect(onClick).toBeCalledTimes(1);
  });
});
