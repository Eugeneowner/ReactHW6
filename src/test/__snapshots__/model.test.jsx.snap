// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Check snapshot component without btn upper 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modalContainer"
      >
        <div
          class="modal"
        >
          <div
            class="upPanel"
          >
            <p
              class="textBold"
            />
          </div>
          <p
            class="textSmall"
          />
          <div
            class="btnsContainer"
          >
            <button
              class="btn"
            >
              Confirm
            </button>
            <button
              class="btn"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modalContainer"
    >
      <div
        class="modal"
      >
        <div
          class="upPanel"
        >
          <p
            class="textBold"
          />
        </div>
        <p
          class="textSmall"
        />
        <div
          class="btnsContainer"
        >
          <button
            class="btn"
          >
            Confirm
          </button>
          <button
            class="btn"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal component Check snapshot component without close 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modalContainer"
      >
        <div
          class="modal"
        >
          <div
            class="upPanel"
          >
            <p
              class="textBold"
            />
          </div>
          <p
            class="textSmall"
          />
          <div
            class="btnsContainer"
          >
            <button
              class="btn"
            >
              Confirm
            </button>
            <button
              class="btn"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modalContainer"
    >
      <div
        class="modal"
      >
        <div
          class="upPanel"
        >
          <p
            class="textBold"
          />
        </div>
        <p
          class="textSmall"
        />
        <div
          class="btnsContainer"
        >
          <button
            class="btn"
          >
            Confirm
          </button>
          <button
            class="btn"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal component Check snapshot with props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modalContainer"
      >
        <div
          class="modal"
        >
          <div
            class="upPanel"
          >
            <p
              class="textBold"
            />
          </div>
          <p
            class="textSmall"
          />
          <div
            class="btnsContainer"
          >
            <button
              class="btn"
            >
              Confirm
            </button>
            <button
              class="btn"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modalContainer"
    >
      <div
        class="modal"
      >
        <div
          class="upPanel"
        >
          <p
            class="textBold"
          />
        </div>
        <p
          class="textSmall"
        />
        <div
          class="btnsContainer"
        >
          <button
            class="btn"
          >
            Confirm
          </button>
          <button
            class="btn"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal component Check snapshot without props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modalContainer"
      >
        <div
          class="modal"
        >
          <div
            class="upPanel"
          >
            <p
              class="textBold"
            />
          </div>
          <p
            class="textSmall"
          />
          <div
            class="btnsContainer"
          >
            <button
              class="btn"
            >
              Confirm
            </button>
            <button
              class="btn"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modalContainer"
    >
      <div
        class="modal"
      >
        <div
          class="upPanel"
        >
          <p
            class="textBold"
          />
        </div>
        <p
          class="textSmall"
        />
        <div
          class="btnsContainer"
        >
          <button
            class="btn"
          >
            Confirm
          </button>
          <button
            class="btn"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
